### ISSUE TITLE
storage provisioning broken after node failure

### ISSUE THREAD

**Issue by justinrush** [longhorn/longhorn#1613]
* docker: 19.3.6
* k8s: 1.16.6
* rancher: 2.3.5
* deployment: on-prem baremetal (Ubuntu 18.04)

I have 9 nodes in my cluster, one died. When it died, all the existing PVs went readonly. I was able to just restart some of those pods and they started working, but one deployment was totally hosed. I tried to upgrade from 0.8 to 1.0, which worked for all but the pod on the failed host, but that didn't fix the broken deployment; it was still getting errors saying:
```
Warning  FailedAttachVolume  117s       attachdetach-controller              AttachVolume.Attach failed for volume "pvc-89b5c7b3-150b-42fe-b431-591e80bdfbec" : attachment timeout for volume pvc-89b5c7b3-150b-42fe-b431-591e80bdfbec

```

I used RKE to remove the failed node and re-ran the longhorn upgrade. The CSI-attacher has this in the logs:
```
time="2020-07-20T12:47:34Z" level=debug msg="Deploying CSI driver"
time="2020-07-20T12:47:34Z" level=debug msg="proc cmdline detection pod discover-proc-kubelet-cmdline in phase: Pending"
time="2020-07-20T12:47:35Z" level=debug msg="proc cmdline detection pod discover-proc-kubelet-cmdline in phase: Pending"
time="2020-07-20T12:47:36Z" level=debug msg="proc cmdline detection pod discover-proc-kubelet-cmdline in phase: Pending"
time="2020-07-20T12:47:37Z" level=info msg="Proc found: kubelet"
time="2020-07-20T12:47:37Z" level=info msg="Try to find arg [--root-dir] in cmdline: [kubelet --cluster-dns=10.43.0.10 --authentication-token-webhook=true --event-qps=0 --cni-bin-dir=/opt/cni/bin --tls-cipher-suites=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305 --volume-plugin-dir=/usr/libexec/kubernetes/kubelet-plugins/volume/exec --client-ca-file=/etc/kubernetes/ssl/kube-ca.pem --cluster-domain=cluster.local --fail-swap-on=false --kubeconfig=/etc/kubernetes/ssl/kubecfg-kube-node.yaml --cni-conf-dir=/etc/cni/net.d --streaming-connection-idle-timeout=30m --authorization-mode=Webhook --hostname-override=swarth-dok-003.denied.life --pod-infra-container-image=rancher/pause:3.1 --network-plugin=cni --resolv-conf=/etc/resolv.conf --v=2 --anonymous-auth=false --cloud-provider= --root-dir=/var/lib/kubelet --address=0.0.0.0 --make-iptables-util-chains=true --read-only-port=0 --cgroups-per-qos=True --cgroup-driver=cgroupfs --resolv-conf=/run/systemd/resolve/resolv.conf ]"
time="2020-07-20T12:47:37Z" level=info msg="Detected root dir path: /var/lib/kubelet"
time="2020-07-20T12:47:37Z" level=info msg="Upgrading Longhorn related components for CSI v1.1.0"
time="2020-07-20T12:47:37Z" level=debug msg="Detected CSI Driver driver.longhorn.io CSI version v1.0.0 Kubernetes version v1.16.6 has already been deployed"
time="2020-07-20T12:47:37Z" level=debug msg="Detected service csi-attacher CSI version v1.0.0 Kubernetes version v1.16.6 has already been deployed"
time="2020-07-20T12:47:37Z" level=debug msg="Waiting for foreground deletion of deployment csi-attacher"
time="2020-07-20T12:49:38Z" level=fatal msg="Error deploying driver: failed to deploy deployment csi-attacher: failed to cleanup deployment csi-attacher: Foreground deletion of deployment csi-attacher timed out"
```

When I deploy a new chart it creates the PVC and it gets bound to a new PV, however, in the Longhorn UI this PV is shown as detached. I tried to delete the PVs from k8s to try fresh again and they are stuck in a terminating state. It looks like the finalizer on those PVs is: external-attacher/driver-longhorn-io

So basically it seems like my longhorn installation is broken. Is there a way to recover from this or I do i have to delete all of it and start over again?

**Comment by yasker** [comment]
You don't need to delete and recreate everything.

`time="2020-07-20T12:49:38Z" level=fatal msg="Error deploying driver: failed to deploy deployment csi-attacher: failed to cleanup deployment csi-attacher: Foreground deletion of deployment csi-attacher timed out"`

It seems csi-attacher deployment cannot be deleted and updated because one of the node is down. It's similar to #1600 

You can manually delete the `csi-attacher` pod on the stuck node using `kubectl -n longhorn-system delete --force pod <xxxx> `, this should allow the driver deployer to deploy the new deployment for the `csi-attacher`.

**Comment by justinrush** [comment]
The failed node is no longer in the cluster, I removed it

**Comment by yasker** [comment]
What would happen if you run `kubectl -n longhorn-system delete deployment csi-attacher` ? Would it delete the `csi-attacher` deployment?

**Comment by justinrush** [comment]
I don't see why it wouldn't, if I delete that will something else magically recreate it?

**Comment by PhanLe1010** [comment]
The `csi-attacher` would not be magically recreated after you delete `kubectl -n longhorn-system delete deployment csi-attacher`. You can scale down the `longhorn-driver-deployer`deployment to 0. Then scale it up back to 1. The `longhorn-driver-deployer` deployment will try to recreate `csi-attacher` deployment

**Comment by justinrush** [comment]
Thanks, force deleting the pods in the various csi-* deployments seems to have unstuck them. When I deleted them, something deleted the entire deployment and re-created it and now things are working again.

Is it expected behavior that on a node failure I will need to manually and forcefully delete these pods?

**Comment by yasker** [comment]
@justinrush it's not. We're still trying to reproduce the issue to understand what happened exactly.

**Comment by yasker** [comment]
Keep this open

**Comment by innobead** [comment]
remove stale due to the incorrect exempt label
